### PR TITLE
feat: implement theme changing feature to sync with parent app

### PR DESCRIPTION
**Pull request by newweborder007**
This PR adds support for receiving and applying theme changes broadcasted from the parent application, ensuring a consistent look and feel across the entire application suite.

### Changes
- Updated the child app to listen for theme change events from the parent app via window events.
- Implemented logic to dynamically update the app's theme based on changes broadcasted from the parent app.

### CODE REVIEW

**Review by wwills2 — APPROVED** [review]

File: src/components/blocks/LeftNav.js [hunk 1]
```diff
@@ -80,6 +79,30 @@ const LeftNav = withTheme(({ children, theme }) => {
   const { isGovernance } = useSelector(state => state.climateWarehouse);
   const dispatch = useDispatch();
   const [myOrgUid, isMyOrgPending] = useSelector(store => getHomeOrg(store));
+  const [colors, setColors] = useState({
+    topBarBgColor: undefined,
+    leftNavHighlightColor: undefined,
+    leftNavBgColor: undefined,
+    leftNavTextColor: undefined,
+  });
+  function notifyParentWhenLeftNavLoaded() {
```
Comment (wwills2) [inline src/components/blocks/LeftNav.js:88]: lets use an arrow function

**Review by wwills2 — CHANGES_REQUESTED** [review]

File: src/components/blocks/LeftNav.js (hunk 1 above)
Comment (wwills2) [inline src/components/blocks/LeftNav.js:88]: lets use an arrow function

**Review by wwills2 — CHANGES_REQUESTED** [review]

File: src/components/blocks/LeftNav.js (hunk 1 above)
Comment (wwills2) [inline src/components/blocks/LeftNav.js:88]: please use an arrow function

**Review by wwills2 — CHANGES_REQUESTED** [review]

File: src/App.js [hunk 2]
```diff
@@ -58,7 +77,7 @@ const App = () => {
   }
 
   return (
-    <ThemeProvider theme={theme}>
+    <ThemeProvider theme={appStore.customTheme}>
```
[the remark this answers is not in the document]
Comment (newweborder007) [reply src/App.js:80]: the content of theme.js is the inital state of appStore.customTheme, so it's already working like this.

**Comment by newweborder007** [comment]
Hello @wwills2,

Changes have been implemented.
- I removed the conditional rendering of colors, and used default colors.
- Redefined the structure of theme.json in parent app, We can now change any color of theme using parent theme.json.

Note: appStore.theme already have static theme in it's initial state so it's good to go.

**Review by wwills2 — APPROVED** [review]

**Pull request merged**